### PR TITLE
CLC-6180 Handle malformed data in textbook feeds

### PR DESCRIPTION
**Pull request by paulkerschen**
https://jira.ets.berkeley.edu/jira/browse/CLC-6180

### CODE REVIEW

File: app/models/textbooks/proxy.rb [hunk 1]
```diff
@@ -34,7 +36,12 @@ def google_book(isbn)
     end
 
     def process_material(material, sections_with_books)
-      isbn = material['ean']
+      begin
+        isbn = Integer(material['ean'], 10).to_s
```
Comment (johncrossman) [inline app/models/textbooks/proxy.rb]: Is 13 digit ISBN a concern?

Comment (paulkerschen) [inline app/models/textbooks/proxy.rb]: 10 is base 10 actually...

File: app/models/textbooks/proxy.rb [hunk 2]
```diff
@@ -148,13 +154,10 @@ def request_bookstore_list(section_numbers)
...
-      path = Rails.root.join('fixtures', 'json', "textbooks-#{@slug}-#{@dept}-#{@course_catalog}-#{section_numbers.join('-')}.json").to_s
-      logger.info "Fake = #{@fake}, getting textbook data from JSON fixture file #{path}"
-      unless File.exists?(path)
-        raise Errors::ProxyError.new("Unrecorded textbook response #{path}")
-      end
-      safe_json File.read(path)
+    private
+
+    def mock_json
+      read_file('fixtures', 'json', 'textbooks-fall-2014-POL SCI-109G-001.json')
```
Comment (johncrossman) [inline app/models/textbooks/proxy.rb]: I see that the space in filename is legit. Rename the file as part of PR?

Comment (paulkerschen) [inline app/models/textbooks/proxy.rb]: Why not, @johncrossman. It does look weird.

**Comment by johncrossman** [comment]
Looks good – thanks for clarification on Integer class. :)  I'll merge once Travis goes green.

**Pull request merged**